### DISCUSSION TITLE
Closing the loop on stepper motors

### DISCUSSION THREAD

**Discussion by Squeack** [terjeio/grblHAL#356] [Q&A]
Stepper motor control is open-loop and cheap. Servos are closed-loop and expensive. I am planning on partly closing the loop on my stepper driven system by feeding the step and directions outputs from the GrblHAL controller (Teensy based breakout board) to both the stepper drivers and to a separate position tracker based on a Raspberry Pi Pico. The three movement leadscrews will be linked via timing belts to 200PPR rotary encoders which will feed back to the Pico. After allowing a ratio for micro-stepping the Pico will then raise an error if the driver instructions and position feedback do not match within a given tolerance.
This has to be communicated to the GrblHAL controller somehow. What would be best? Triggering the door open switch? Does GrblHAL react to the limit switch inputs for the A or B axis even if compiled to handle only three axes?

**Comment by terjeio** [comment]
>  What would be best?

Use the motor fault signal available in [newer versions](https://github.com/grblHAL/iMXRT1062)? It may require a small plugin or at least a modified board map file.

> Does GrblHAL react to the limit switch inputs for the A or B axis even if compiled to handle only three axes?

No. In the latest version unused pins can be repurposed and also made available for plugin code.

Do you compile yourself?

Again, please continue this discusson at the new repo.

**Comment by Squeack** [comment]
Thanks Terjeio. Moved to https://github.com/grblHAL/core/discussions/110
